### PR TITLE
UI flow

### PR DESCRIPTION
**Pull request by harryhcs**
Added new pages:

1. Landing / Welcome page (navigate to list of tradesmen currently on our system)
2. Professions page
3. Tradesmen page
4. Tradesman page

- Added a way to use "dynamic roots" like `/tradesman/:id`

TODO: Upload profile pic, add description to tradesmen, add form to for tradesmen to add themselfs to our system (this is added with a vetted bool field set to false or something like that(

### CODE REVIEW

File: experiments/rsc/src/worker.tsx [hunk 1]
```diff
@@ -136,32 +156,51 @@ export default {
...
+        const tradesmanMatch = pathname.match(/^\/tradesman\/(.+)$/);
+        if (tradesmenMatch) {
+          const profession = tradesmenMatch[1];
+          return renderPage(routes["/tradesmen/:profession"], {
+            params: { profession },
+          });
+        }
+        if (tradesmanMatch) {
+          const id = tradesmanMatch[1];
+          return renderPage(routes["/tradesman/:id"], {
```
Comment (justinvdm) [inline experiments/rsc/src/worker.tsx:196]: Nice! @peterp maybe there's an opportunity here now for us to figure out how routing should work and/or how the API should look.

File: experiments/rsc/src/worker.tsx [hunk 2]
```diff
@@ -8,11 +8,25 @@ import { renderToRscStream } from "./render/renderToRscStream";
...
+  "/welcome": WelcomePage,
   "/admin": AdminPage,
+  "/tradesmen/:profession": (props: { params: { profession: string } }) => (
+    <TradesmenPage profession={props.params.profession} />
+  ),
+  "/tradesman/:id": (props: { params: { id: number } }) => (
+    <TradesmanPage id={props.params.id} />
+  ),
+  "/professions": ProfessionsPage,
+  "/add-tradesman": AddTradesmanPage,
```
Comment (justinvdm) [inline experiments/rsc/src/worker.tsx:29]: Nice API!

Note for myself that I need to investigate if there's a way we can lazy-load modules in cloudflare, we'll probably need this for routing.

File: experiments/rsc/src/app/AdminPage.tsx [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 
 export default async function AdminPage() {
   let tradesmen = await db
     .selectFrom("Tradesman")
-    .select(["id", "name", "cellnumber", "profession"])
+    .select(["id", "name", "cellnumber", "profession", "profilePicture"])
     .execute();
+
+  const files = await R2Storage.listFiles();
+  console.log(files);
```
Comment (justinvdm) [inline experiments/rsc/src/app/AdminPage.tsx:13]: I guess this was a debugging line?

**Review by justinvdm — APPROVED** [review]
Nice!

File: experiments/rsc/src/app/AdminPage.tsx (hunk 3 above)
Comment (harryhcs) [reply experiments/rsc/src/app/AdminPage.tsx:13]: yup, will clean this up this morning

**Pull request merged**